core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java

```diff
@@ -147,25 +148,27 @@ public synchronized OutputStream createCompressionStream(OutputStream downStream
     },
 
     GZ(COMPRESSION_GZ) {
-      private transient DefaultCodec codec;
+      private transient AtomicReference<DefaultCodec> codec = new AtomicReference<DefaultCodec>();
 
       @Override
-      synchronized CompressionCodec getCodec() {
-        if (codec == null) {
-          codec = new DefaultCodec();
-          codec.setConf(conf);
+      CompressionCodec getCodec() {
+        DefaultCodec resultCodec = codec.get();
+        if (null == resultCodec) {
+          DefaultCodec newCodec = new DefaultCodec();
```

Comment:
If a bunch of threads all got to this point at the same time, they would all create a new DefaultCodec. Only one will succeed on the compareAndSet below, but the code is still potentially creating many DefaultCodec instances which will get garbage collected.
Since isSupported returns true, I'm wondering if it's possible to simply initialize the member variable at construction time, and then the AtomicReference wouldn't be needed either? But maybe I'm missing something obvious.